src/cdrizzlebox.c

```diff
@@ -26,18 +26,21 @@ inline_macro static int
 update_data(struct driz_param_t* p, const integer_t ii, const integer_t jj,
             const float d, const float vc, const float dow) {
 
-  const double vc_plus_dow = vc + dow;
+  double vc_plus_dow;
   
-  /* Just a simple calculation without logical tests */
-  if (vc == 0.0) {
+  if (dow == 0.0f) return 0;
+  
+  vc_plus_dow = vc + dow;
```

Comment:
If dow is zero, does this function exit and thus never reach lines 54-59 (particularly line 58)? Is it safe to assume that the data will have already been updated for vc (i.e. there is no situation that skipping line 58 here won't be an issue?))?

Reply:
Good question. The idea in my fix (this PR) is that if a pixel - an _input_ one - has 0 weight (when considering its contribution to an output pixel) and thus it _is not_ contributing to the _output_ pixel at all, then why do we need to update output pixel at all or the total weight (lines 54-59) of that output pixel? Nothing changes when "dropped flux" has 0 weight...

If you look at `cdrizzlebox.c`, you will see that `vc` is simply the current (before "this" update) weight of an output pixel:
```c
vc = get_pixel(p->output_counts, ii, jj);
```

If the flux that we are going to drop onto that output pixel has a non-zero weight, then we should update that weight (and output pixel value). But if the dropped flux is not contributing to the output pixel I do not think its weight should be modified.

If an output fixel has no flux at all in it (because no input pixel contributed to this output pixel) - its weight will be 0 before and after `update_data()` assuming that during this last update dropped flux has also 0 weight (`dow==0`).

Reply:
The mistake in the previous logic (IMO) was that it was testing to see whether output image pixel had a contribution from anything before (`if (vc == 0.0)`) and if nothing ever contributed to that output pixel then the old code would drop input flux regardless of its weight simply _by assigning_ this flux to the output pixel - see the `"then"` block after `if (vc == 0.0)`.

Reply:
Interestingly, with the old code, when it would drop flux with 0 weight onto an output pixel with pre-existing weight of 0, then `vc_plus_dow` would be `0 + 0` and so output pixel weight assigned on lines 54-59 would still be 0. And so when another input image would be added (drizzled) to the ouput, it would just overwrite previous value in those 0-weight regions.